python/pyarrow/io.pxi

```diff
@@ -1053,6 +1062,14 @@ cdef class Buffer(_Weakrefable):
     def __len__(self):
         return self.size
 
+    def __repr__(self):
+        name = f"{self.__class__.__module__}.{self.__class__.__name__}"
+        return (f"{name}("
+                f"{hex(id(self))}, "
```

Comment:
Ah, for Buffers, I would actually be interested in the address of the data itself (so `hex(self.address)`)